pkg/controller/external_nofork.go

```diff
@@ -626,18 +627,17 @@ func getIgnoredFieldsArray(format string, forProvider, initProvider []any) []str
 	ignored := []string{}
 	for i := range initProvider {
 		// Construct the full field path with array index and prefix.
-		fieldPath := fmt.Sprintf("%s[%d]", format, i)
+		fieldPath := fmt.Sprintf("%s.%d", format, i)
 		if i < len(forProvider) {
 			if _, ok := initProvider[i].(map[string]any); ok {
 				ignored = append(ignored, getIgnoredFieldsMap(fieldPath+".%s", forProvider[i].(map[string]any), initProvider[i].(map[string]any))...)
 			}
 			if _, ok := initProvider[i].([]any); ok {
-				ignored = append(ignored, getIgnoredFieldsArray(fieldPath+"%s", forProvider[i].([]any), initProvider[i].([]any))...)
```

Comment:
Looks like although the first argument is named as `format`, it's not actually a format parameter. Could you please open a bug regarding this?

Reply:
Done: [`ignore_changes` formatting is wrong for list of lists](https://github.com/crossplane/upjet/issues/295)